ignite/config/plugins/config.go

```diff
@@ -42,18 +42,55 @@ type Plugin struct {
 	Global bool `yaml:"-"`
 }
 
+// RemoveDuplicates takes a list of Plugins and returns a new list with only unique values.
+// Local plugins take precedence over global plugins if duplicate paths exist.
+// Duplicates are compared regardless of version.
+func RemoveDuplicates(plugins []Plugin) (unique []Plugin) {
+	// struct to track plugin configs
+	type check struct {
+		hasPath    bool
+		global    bool
+		prevIndex int
+	}
+
+	keys := make(map[string]check)
+	for i, plugin := range plugins {
+		c := keys[plugin.SubPath()]
+		if !c.hasPath {
+			keys[plugin.SubPath()] = check{
+				hasPath:    true,
+				global:    plugin.Global,
+				prevIndex: i,
+			}
+			unique = append(unique, plugin)
+		} else if c.hasPath && !plugin.Global && c.global { // overwrite global plugin if local duplicate exists
+			unique[c.prevIndex] = plugin
+		}
+	}
+
+	return unique
+}
+
+// HasPath verifies if a plugin has the given path regardless of version.
+// Example:
+// github.com/foo/bar@v1 and github.com/foo/bar@v2 have the same path so "true"
+// will be returned.
 func (p Plugin) HasPath(path string) bool {
 	if path == "" {
 		return false
 	}
 	if p.Path == path {
 		return true
 	}
-	pluginPath := strings.Split(p.Path, "@")[0]
+	pluginPath := p.SubPath()
 	path = strings.Split(path, "@")[0]
 	return pluginPath == path
 }
 
+func (p Plugin) SubPath() string {
```

Comment:
What about `CanonicalPath` or `BasePath` instead ?

Reply:
Much better.  I didn't like this name.  Thanks!

Reply:
https://github.com/ignite/cli/pull/3293/commits/58dde352f4936ec1e14330aca7c7155a71468b86

Reply:
:pray: